crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -1358,6 +1362,189 @@ where
                     );
                 }
             }
+            EnrichedEntryHeader::GetPromise { is_completed, .. } => {
+                if !is_completed {
+                    let_assert!(
+                        Entry::GetPromise(GetPromiseEntry { key, .. }) =
+                            journal_entry.deserialize_entry_ref::<Codec>()?
+                    );
+
+                    if let Some(service_id) =
+                        invocation_metadata.invocation_target.as_keyed_service_id()
+                    {
+                        // Load state and write completion
+                        let promise_metadata = state.get_promise(&service_id, &key).await?;
+
+                        match promise_metadata {
+                            Some(PromiseMetadata {
+                                state: PromiseState::Completed(result),
+                            }) => {
+                                // Result is already available
+                                let completion_result: CompletionResult = result.into();
+                                Codec::write_completion(
+                                    &mut journal_entry,
+                                    completion_result.clone(),
+                                )?;
+
+                                // Forward completion
+                                effects.forward_completion(
+                                    invocation_id,
+                                    Completion::new(entry_index, completion_result),
+                                );
+                            }
+                            Some(PromiseMetadata {
+                                state: PromiseState::NotCompleted(mut v),
+                            }) => {
+                                v.push(JournalEntryId::from_parts(invocation_id, entry_index));
+                                effects.put_promise(
+                                    service_id,
+                                    key,
+                                    PromiseMetadata {
+                                        state: PromiseState::NotCompleted(v),
+                                    },
+                                )
+                            }
+                            None => effects.put_promise(
+                                service_id,
+                                key,
+                                PromiseMetadata {
+                                    state: PromiseState::NotCompleted(vec![
+                                        JournalEntryId::from_parts(invocation_id, entry_index),
+                                    ]),
+                                },
+                            ),
+                        }
+                    } else {
+                        warn!(
+                            "Trying to process entry {} for a target that has no promises",
+                            journal_entry.header().as_entry_type()
+                        );
```

Comment:
A bit unrelated. I am not sure about the warn level here because technically this situation can easily happen when a user writes an incorrect handler. However, this does not mean that the runtime is in anyway in danger which I would usually associate with a log message on warn level (e.g. disk space is running full, not able to connect to the cluster controller, etc.).

Reply:
> A bit unrelated. I am not sure about the warn level here because technically this situation can easily happen when a user writes an incorrect handler. 

This can happen only in case the SDK is buggy and exposes the promise apis to non-workflow `Context`, this is a similar case to creating GetState inside a regular service.